src/db/mod.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+/// The names of the collections in the MongoDB database.
+mod collections;
+mod error;
+pub use error::MongoDbError;
+use inx::proto::{Message, Milestone};
+use mongodb::{
+    bson,
+    bson::{doc, Document},
+    options::{ClientOptions, Credential},
+    Client,
+};
+use serde::{Deserialize, Serialize};
+
+/// Name of the MongoDB database.
+pub const DB_NAME: &str = "chronicle-test";
+
+/// A builder to establish a connection to the database.
+#[must_use]
+#[derive(PartialEq, Eq, Debug, Serialize, Deserialize)]
+pub struct MongoConfig {
```

Comment:
That's a bit different to what we did in Bee, but it seems to be on purpose, right?

What about the pattern `let mongo_db = MongoConfig::build().with_a().with_b().finish();`?

I really like that pattern. Or is this just temporary and supposed to be changed by me in the config PR?

Reply:
Yeah, I figured these will change anyway so I didn't want to bother separating `{Inx|MongoDb}Config` and `{Inx|MongoDb}ConfigBuilder`. Would that be ok, for you?